<!DOCTYPE html>
<script src="../../resources/testharness.js"></script>
<script src="../../resources/testharnessreport.js"></script>
<script src="../../resources/bluetooth/bluetooth-helpers.js"></script>
<script>
'use strict';
promise_test(() => {
  return setBluetoothFakeAdapter('TwoHeartRateServicesAdapter')
    .then(() => requestDeviceWithKeyDown({
      filters: [{services: ['heart_rate']}],
      optionalServices: ['generic_access']}))
    .then(device => device.gatt.connect())
    .then(gattServer => {
      let promise = assert_promise_rejects_with_message(
        gattServer.getPrimaryServices(),
        new DOMException('GATT Server disconnected while retrieving services.',
                         'NetworkError'));
      gattServer.disconnect();
      return promise;
    });
}, 'disconnect() called during getPrimaryServices. Reject with NetworkError.');
</script>
